ci/Makefile

```diff
@@ -15,7 +25,7 @@ update-conda-env:
 build-hail-ci:
 	cd ../ && docker build . -t hail-ci -f ci/Dockerfile
 
-push-hail-ci: IMAGE = gcr.io/$(PROJECT)/hail-ci:$(shell docker images -q --no-trunc hail-ci | head -n 1 | sed -e 's,[^:]*:,,')
+push-hail-ci: IMAGE = gcr.io/${PROJECT}/hail-ci:$(shell docker images -q --no-trunc hail-ci | head -n 1 | sed -e 's,[^:]*:,,')
```

Comment:
are these two forms semantically equivalent in Make?

Reply:
Yep! I made the change for consistency, so that we could avoid confusing future readers as to our intentions (I needed to double check on why ${ } was used vs $( ) for substitution, thought it was something make specific).

"To substitute a variable’s value, write a dollar sign followed by the name of the variable in parentheses or braces: either `‘$(foo)’`or `‘${foo}’` is a valid reference to the variable foo."

https://www.gnu.org/software/make/manual/html_node/Reference.html
https://stackoverflow.com/questions/25185607/whats-the-difference-between-parenthesis-and-curly-bracket-syntax-in-ma

I found it interesting, since they are 100% different in the context we typically use them, as a right hand expression to be evaluated during assignment.

```sh
AlexKotlar:ci alexkotlar$ test=$(gcloud compute instances list)
AlexKotlar:ci alexkotlar$ echo $test
NAME ZONE MACHINE_TYPE PREEMPTIBLE INTERNAL_IP EXTERNAL_IP STATUS instance-1 us-east1-b f1-micro 10.142.0.2 35.237.80.71 RUNNING

AlexKotlar:ci alexkotlar$ test=${gcloud compute instances list}
-bash: ${gcloud compute instances list}: bad substitution
```

[Here parenthesis act as a subshell](https://unix.stackexchange.com/questions/26063/how-are-parentheses-interpreted-at-the-command-line)

Reply:
I've done a little digging and it seems like while $(VAR) and ${VAR} are equivalent in both POSIX and GNU Make, they weren't always semantically identical in both versions.

@danking 	@cseed 	- is there a reason we used $() in the first place? convention?

Reply:
Mixing syntax in one expression is apparently discouraged; I used the form seen in the global variables because it was testable in .sh directly. Will update, and add := for single evaluation.

Reply:
I think `$(FOO)` is traditional in Makefiles. In bash, `$FOO` is probably most traditional, but I prefer `${FOO}` since it doesn't blow up if you remove word-separating tokens around it.

Reply:
I honestly was unaware `${FOO}` was acceptable in GNU Make.

Reply:
A bit more reading:
1) Oreilly claims $( ) are more idiomatic. However, they have an obvious disadvantage: when testing sections of the makefile, you would need to always replace $( ) with ${ }.

https://www.oreilly.com/openbook/make3/book/ch03.pdf 

2) BSD uses ${ } consistently, both in manual, and apparently in BSD packages
https://stackoverflow.com/questions/37971448/using-makefile-variable-in-sed-command
https://www.freebsd.org/cgi/man.cgi?make(1)
https://man.openbsd.org/make

We should probably choose one of $()/${} and stick with it, and I don't mind one over the other if there are incompatibilities with OSs we care about as @tpoterba suggests there may be.

Reply:
I'm totally fine with curly braces! Paste-ability into bash does seem nice.

Reply:
I think we could prefer `$()` in Makefile since we already do in our other Makefiles. If we want to switch to `${}`, let's do a separate PR that changes that for every makefile in this project.

Reply:
I agree, will change to $( )